src/main/java/com/coveo/pushapiclient/DocumentUploadQueue.java

```diff
@@ -1,21 +1,105 @@
 package com.coveo.pushapiclient;
 
 import java.io.IOException;
+import java.util.ArrayList;
 
-// TODO: LENS-851 - Make public
+/**
+ * Represents a queue for uploading documents using a specified upload strategy
+ */
 class DocumentUploadQueue {
     private final UploadStrategy uploader;
+    private final int maxQueueSize = 5 * 1024 * 1024;
+    private ArrayList<DocumentBuilder> documentToAddList;
+    private ArrayList<DeleteDocument> documentToDeleteList;
```

Comment:
Why use `ArrayList` instead of a Queue (or any other `Collection`)?

https://docs.oracle.com/javase/8/docs/api/java/util/Queue.html

Reply:
[LinkedList](https://docs.oracle.com/javase/8/docs/api/java/util/LinkedList.html) are slower in most cases and  clearing the entire list requires traversing and removing each element individually as opposed to ArrayList which set all elements to null.

I though of [HashSets](https://docs.oracle.com/javase/8/docs/api/java/util/HashSet.html) for the following reasons:
* We don't want duplicate entries in the queue
* Order is not important
* constant-time performance for most operations

However, even if they are better at finding an element by value and by clearing all the elements (`O(1)` vs `O(n)`), I don't know if they are the best data structure to be used as a "container" to store elements memory-wise.

I plan to do performance testing at a later with large amount of data to ensure the queue is the most efficient it can be.

I think, ArrayList is a good compromise for now.

Reply:
Also, 
Most of the exposed data structures that deal with documents are ArrayList. Switching to another data structure without solid reason would cause breaking changes elsewhere.

Reply:
Fair enough! I just wanted to be sure it have been thought of/consider!